pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -320,3 +326,109 @@ func (nm *NodeManager) convertK8sUUIDtoNormal(k8sUUID string) string {
 		k8sUUID[24:36])
 	return strings.ToLower(uuid)
 }
+
+// ExportNodes transforms the NodeInfoList to []*pb.Node
+func (nm *NodeManager) ExportNodes(vcenter string, datacenter string, nodeList *[]*pb.Node) error {
+	nm.nodeInfoLock.Lock()
+
+	if vcenter != "" && datacenter != "" {
+		dc, err := nm.FindDatacenterInfoInVCList(vcenter, datacenter)
+		if err != nil {
+			nm.nodeInfoLock.Unlock()
+			return err
+		}
+
+		nm.datacenterToNodeList(dc.vmList, nodeList)
+	} else if vcenter != "" {
+		if nm.vcList[vcenter] == nil {
+			nm.nodeInfoLock.Unlock()
+			return ErrVCenterNotFound
+		}
+
+		for _, dc := range nm.vcList[vcenter].dcList {
+			nm.datacenterToNodeList(dc.vmList, nodeList)
+		}
+	} else {
+		for _, vc := range nm.vcList {
+			for _, dc := range vc.dcList {
+				nm.datacenterToNodeList(dc.vmList, nodeList)
+			}
+		}
+	}
```

Comment:
disclaimer: I'm nit picking here...

I would suggest to use `switch` instead of `if-else` chains, they're easier to read and more idiomatic.

Reply:
Good feedback. I agree. Will make changes.

Reply:
I just took a look and the first if statement is conditional on 2 parameters.